### PR TITLE
[WIP][CR] Add constants for easily changing the size of the reality bubble at compile time.

### PR DESCRIPTION
**Pull request by mrkybe**
They should really be documented better, but I don't have enough of an understanding of the various parts to do it.

### CODE REVIEW

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -890,7 +890,7 @@ void activity_handlers::firstaid_finish( player_activity *act, player *p )
 static void rod_fish( player *p, int sSkillLevel, int fishChance )
 {
    if( sSkillLevel > fishChance ) {
-        std::vector<monster *> fishables = g->get_fishable(60); //get the nearby fish list.
+        std::vector<monster *> fishables = g->get_fishable(DISTANCE_FROM_CENTER); //get the nearby fish list.
```
Comment (kevingranade) [inline src/activity_handlers.cpp]: This is range at which you can attract fish to a lure, 60 already seems a bit overboard...

File: src/computer.cpp [hunk 2]
```diff
@@ -512,10 +512,10 @@ void computer::activate_function( computer_action action )
     break;
 
     case COMPACT_MAP_SEWER: {
-        g->u.moves -= 30;
```
Comment (kevingranade) [inline src/computer.cpp]: This is the cost of reading the map, not distance related.

File: src/map.cpp [hunk 3]
```diff
@@ -5695,18 +5695,26 @@ const visibility_variables &map::get_visibility_variables_cache() const {
 
 lit_level map::apparent_light_at( const tripoint &p, const visibility_variables &cache ) const {
     const int dist = rl_dist( g->u.pos(), p );
+    
+    // Being outside of the world bubble overrides Clairvoyance.
```
Comment (kevingranade) [inline src/map.cpp]: This shouldn't be necessary, wasn't the fix for this in handle_mouseview?

File: src/map.cpp [hunk 4]
```diff
@@ -5695,18 +5695,26 @@ const visibility_variables &map::get_visibility_variables_cache() const {
...
         return LL_BRIGHT;
     }
+
     const auto &map_cache = get_cache_ref(p.z);
     bool obstructed = map_cache.seen_cache[p.x][p.y] <= LIGHT_TRANSPARENCY_SOLID + 0.1;
     const float apparent_light = map_cache.seen_cache[p.x][p.y] * map_cache.lm[p.x][p.y];
 
     // Unimpaired range is an override to strictly limit vision range based on various conditions,
     // but the player can still see light sources.
-    if( dist > g->u.unimpaired_range() ) {
```
Comment (kevingranade) [inline src/map.cpp]: Unnecessary change.

File: src/monattack.cpp [hunk 5]
```diff
@@ -4512,7 +4512,7 @@ bool mattack::grenadier(monster *const z)
     if (z->attitude_to( *target ) == Creature::A_FRIENDLY) {
         return false;
     }
-    int ret = grenade_helper(z, target, 30, 60, grenades);
+    int ret = grenade_helper(z, target, 30, DISTANCE_FROM_CENTER, grenades);
```
Comment (kevingranade) [inline src/monattack.cpp]: I suspect this is a range, not a position, it doesn't necessarily need updating.

File: src/player.cpp [hunk 6]
```diff
@@ -2756,8 +2756,8 @@ float player::active_light() const
 
     lumination = ( float )maxlum;
 
-    if( lumination < 60 && has_active_bionic( bio_flashlight ) ) {
-        lumination = 60;
```
Comment (kevingranade) [inline src/player.cpp]: This is a lighting level, not a distance.

File: src/sounds.cpp [hunk 7]
```diff
@@ -175,7 +175,7 @@ int get_signal_for_hordes( const centroid &centr )
     //Volume in  tiles. Signal for hordes in submaps
     //modify vol using weather vol.Weather can reduce monster hearing
     const int vol = centr.volume - weather_data( g->weather ).sound_attn;
-    const int min_vol_cap = 60; //Hordes can't hear volume lower than this
+    const int min_vol_cap = DISTANCE_FROM_CENTER; //Hordes can't hear volume lower than this
```
Comment (kevingranade) [inline src/sounds.cpp]: Not map related.

File: src/weather_data.cpp [hunk 8]
```diff
@@ -42,7 +42,7 @@ weather_datum const weather_data( weather_type const type )
      * Weather types data definition.
      * Name, color in UI, ranged penalty, sight penalty,
      * light modifier, sound attenuation, warn player?
-     * Note light modifier assumes baseline of DAYLIGHT_LEVEL at 60
```
Comment (kevingranade) [inline src/weather_data.cpp]: Again, light level, not distance.

**Comment by BevapDin** [comment]
Please don't add macros, add constants as `static constexpr`.

File: src/game_constants.h [hunk 9]
```diff
@@ -88,6 +88,18 @@
...
+#define PLAYER_VIEW_DISTANCE 60.0f
+
+/** Player view distance INTEGER */
+#define PLAYER_VIEW_DISTANCE_INT 60
+
+/** Kind of related... but not to view distance exactly? Changing this would change balance of things I think? */
+#define DISTANCE_FROM_CENTER 60
+
+/** Was used based off lighting but I'm not sure?*/
+#define LIGHT_SIXTY_F 60.0f
```
Comment (Coolthulhu) [inline src/game_constants.h]: This won't work right. You'd need to alter air transparency instead, otherwise you'll get a ton of side effects.

Comment (BevapDin) [inline src/game_constants.h]: If you put the (current) value of the constant into its name, you might as well use the value directly. As soon as somebody changes the value, the name will be misleading.

Regarding two constants for the same value (int and float): define just one of them. And if you really need the other one (note: the compiler will convert the values as it needs it anyway), you should define it with a reference to the first one, e.g.
```C++
constexpr int DIST = 60;
constexpr float DIST_FLT = DIST;
```

But it's better to use the integer constant all the time, the compiler will automatically cast it *at compile time* when needed.

File: src/map.cpp [hunk 10]
```diff
@@ -5695,17 +5695,24 @@ const visibility_variables &map::get_visibility_variables_cache() const {
 
 lit_level map::apparent_light_at( const tripoint &p, const visibility_variables &cache ) const {
     const int dist = rl_dist( g->u.pos(), p );
+    
+    // Being outside of the world bubble overrides Clairvoyance.
+    if(p.x < 0 || p.y < 0) {
```
Comment (BevapDin) [inline src/map.cpp]: We have `map::inbounds` for this, which works with `tripoint` directly, checks the `z` component as well and checks for the upper bound as well.

Comment (Coolthulhu) [reply src/map.cpp]: It's a bad idea to have this check here, though.
`apparent_light_at` is used a lot and is one of the most expensive functions, according to profiler.

File: src/activity_handlers.cpp (hunk 1 above)
Comment (mrkybe) [reply src/activity_handlers.cpp]: Renamed to fishing_distance.

File: src/monattack.cpp (hunk 5 above)
Comment (mrkybe) [reply src/monattack.cpp]: Upon further examination, this was a moves cost.  Returned back to '60'.

File: src/game_constants.h (hunk 9 above)
Comment (mrkybe) [reply src/game_constants.h]: Changed to using `constexpr`.  My goal was to replace hardcoded values with constants while being certain that it wouldn't have any side-effects.  I wanted to avoid side effects related to how integer division works vs floats, so I kept the types the same.

File: src/map.cpp (hunk 3 above)
Comment (mrkybe) [reply src/map.cpp]: Finally getting back to this PR.  So apparently this function isn't only called from handle_mouseview, which is why I ended up handling it here instead of in handle_mouseview.

Comment (Coolthulhu) [reply src/map.cpp]: It is used heavily in `map::update_visibility_cache`, which is called on the entire bubble and only bubble (ie. it doesn't ever cross boundaries of the bubble).
`update_visibility_cache` is not cheap and `apparent_light_at` is the most expensive part of it.

Comment (mrkybe) [reply src/map.cpp]: `map::inbounds` isn't an expensive check though.  I'm not sure how else to do it.

Comment (Coolthulhu) [reply src/map.cpp]: Extract the check to `print_all_tile_info`, which is the only place where this check matters.

Comment (mrkybe) [reply src/map.cpp]: Ah, I see.  Done.

File: src/game.cpp [hunk 11]
```diff
@@ -7329,7 +7329,13 @@ void game::print_all_tile_info( const tripoint &lp, const catacurses::window &w_
                                 const int last_line, bool draw_terrain_indicators,
                                 const visibility_variables &cache )
 {
-    auto visibility = m.get_visibility( m.apparent_light_at( lp, cache ), cache );
+    visibility_type visibility;
```
Comment (kevingranade) [inline src/game.cpp]: how about:
```
visibility = VIS_HIDDEN
if( g->map_ptr->inbounds( lp ) ) {
    visibility = m.get_visibility( m.apparent_light_at( lp, cache ), cache );
}
```
Just a styling thing (including the spaces around parentheses).

File: src/game_constants.h [hunk 12]
```diff
@@ -88,6 +88,18 @@
...
 #define RANGE_HARD_CAP 60
 
+/** Player view distance FLOAT */
+constexpr float player_view_distance = 60.0f;
+
+/** Player view distance INTEGER */
+constexpr int player_view_distance_int = player_view_distance;
+
+/** Distance to search for fishables from trap/fishing pole/etc */
+constexpr int fishing_distance = 60;
```
Comment (kevingranade) [inline src/game_constants.h]: Please hoist this to a common shared header like iuse_actor.h, no need to make this visible to the entire game.

File: src/game_constants.h [hunk 13]
```diff
@@ -88,6 +88,18 @@
...
+
+/** Light level to use for starting a fire */
+constexpr float min_light_level_for_firestarter = 60.0f;
```
Comment (kevingranade) [inline src/game_constants.h]: Also move to iuse_actor.h, or move them both to iuse_constants.h, just not here.

File: src/game_constants.h (hunk 9 above)
Comment (kevingranade) [reply src/game_constants.h]: Get rid of the int variant please.

File: src/game_constants.h [hunk 14]
```diff
@@ -88,6 +88,18 @@
 /** Maximum range at which ranged attacks can be executed */
 #define RANGE_HARD_CAP 60
 
+/** Player view distance FLOAT */
+constexpr float player_view_distance = 60.0f;
```
Comment (kevingranade) [inline src/game_constants.h]: Please hoist to a header file shared by lightmap.cpp and player.cpp, map.h would work.

**Comment by ZhilkinSerg** [comment]
Feel free to reopen.
